torch/_inductor/cudagraph_trees.py

```diff
@@ -1103,11 +1148,11 @@ def __init__(self, device_index: int):
         # will not be reused; separate recordings would have use the same memory pool, but not
         # the same memory.
 
-        torch.cuda.synchronize()
-        self.stream = torch.cuda.Stream()
-        self.stream.wait_stream(torch.cuda.current_stream())
```

Comment:
weird that this used to work, what's going on here?

Reply:
I'm not sure why this change triggered a failure in multi device scenario - but i added additional tests/run for multi device